### ISSUE TITLE
Navigation.push() got delayed if the source component is a heavy component

### ISSUE THREAD

**Issue by sanghavan** [wix/react-native-navigation#6852]
## 🐛 Bug Report

Navigation.push() got delayed if the source component is a heavy component (E.g. The source component is a long items list, or a section list has many sections data)

## To Reproduce

I can reproduce the issue on playground app.

1. Update the code in `StackScreen.tsx` to render a Flatlist with 2k rows, each row contains <Text /> with row index as value. You can wait for rendering complete.
2. Access into `Layout` bottom tab => Tap on `Stack` button => Tap on `Push` button. Then you will see a bit delay before the slide animation start. Just increasing number of rows for more delay.

## Expected behavior

- No any delay time when pushing to target component

## Actual Behavior

- There is a delay when pushing.

## Your Environment

- The playground app downloaded from release tag `7.6.0`
- React Native Navigation version: `7.6.0`
- React Native version: `0.63.2`
- Platform(s) (iOS, Android, or both?): `iOS`
- Device info (Simulator/Device? OS version? Debug/Release?): Real device iPhone X, Simulator iPhone 8 Plus. OS: iOS 14.3, The both Debug & Release mode

**Comment by sanghavan** [comment]
Also, switching bottom tabs got delayed if the target tab is a heavy component.
Example:
Tab 1 | Tab 2.
Tab 1 is a heavy component. When switching Tab 1 -> Tab 2, it will switch immediately. But Tab 2 -> Tab 1, it got delayed. I believe we don't re-render tabs when switch them.

**Comment by yogevbd** [comment]
Is there any chance you set `push.waitForRender: true`? maybe in your default options?

**Comment by sanghavan** [comment]
> Is there any chance you set `push.waitForRender: true`? maybe in your default options?

No, I set either waitForRender: false or remove it, the issue is still same. Base on my understand, waitForRender is used to wait the **target component** render completed before start animation. So for this issue, `waitForRender` attribute doesn't relate to.
The issue: A pushes to B. A is heavy component, B is very lightweight component. But still got delayed.

**Comment by sanghavan** [comment]
Hi! any updates on this?

**Comment by sintylapse** [comment]
@yogevbd this is still an issue

**Comment by sintylapse** [comment]
The problem fixes if you statically specify width and height of the container view

**Comment by hetansh-Tatvasoft** [comment]
> The problem fixes if you statically specify width and height of the container view

can please you elaborate you answer more, because this is serious issue in my app and i don't have any clue on this...

**Comment by sintylapse** [comment]
> > The problem fixes if you statically specify width and height of the container view
> 
> can please you elaborate you answer more, because this is serious issue in my app and i don't have any clue on this...

hey @hetansh-Tatvasoft try to specify the `width` and `height` of the root `View` of the screen you trying to push
```
import { Dimensions, View } from 'react-native';

function YourScreen() {
  const { width, height } = Dimensions.get('window');

  return (
    <View style={{ width, height }}>
      {/* the content of the screen */}
    </View>
  )
}
```

**Comment by hetansh-Tatvasoft** [comment]
> > > The problem fixes if you statically specify width and height of the container view
> > 
> > can please you elaborate you answer more, because this is serious issue in my app and i don't have any clue on this...
> 
> hey @hetansh-Tatvasoft try to specify the `width` and `height` of the root `View` of the screen you trying to push
> ```
> import { Dimensions, View } from 'react-native';
> 
> function YourScreen() {
>   const { width, height } = Dimensions.get('window');
> 
>   return (
>     <View style={{ width, height }}>
>       {/* the content of the screen */}
>     </View>
>   )
> }
> ```

Thanks for the quick response....🚀🚀🚀

**Comment by phamhungvn** [comment]
> > > > The problem fixes if you statically specify width and height of the container view
> > > 
> > > 
> > > can please you elaborate you answer more, because this is serious issue in my app and i don't have any clue on this...
> > 
> > 
> > hey @hetansh-Tatvasoft try to specify the `width` and `height` of the root `View` of the screen you trying to push
> > ```
> > import { Dimensions, View } from 'react-native';
> > 
> > function YourScreen() {
> >   const { width, height } = Dimensions.get('window');
> > 
> >   return (
> >     <View style={{ width, height }}>
> >       {/* the content of the screen */}
> >     </View>
> >   )
> > }
> > ```
> 
> Thanks for the quick response....🚀🚀🚀

@hetansh-Tatvasoft Did you try this approach? Is it working on your case?

**Comment by hetansh-Tatvasoft** [comment]
> 

@phamhungvn tried this approach but it didn't help, I am using flatlist at the source component and when there is few Items in list it works fine, but when items keeps increasing that causes the navigation on click of a particular item becomes very slow....☹️

note:- I use paging in flat list and my feature requirements don't allow me to remove previously load items to remove from list and hold only few items in focus of the list , so more then often it becomes heavy Flatlist.

**Comment by hetansh-Tatvasoft** [comment]
@Harisene For me navigation was not an issue, I've moved to Shopify/FlashList from flatlist and things are working great since then... 

Package - https://shopify.github.io/flash-list/

**Comment by Harisene** [comment]
@hetansh-Tatvasoft. I'm not using any list in my component. Still having issues when navigating.

**Comment by hetansh-Tatvasoft** [comment]
@Harisene Sorry can't help further It seems like we've different issues, Maybe try [this](https://github.com/wix/react-native-navigation/issues/6852#issuecomment-1128658309) and see if it works.
All the best...

**Comment by anhnguyen123** [comment]
still slow

**Comment by yedidyak** [comment]
Closing this as part of a maintainer cleanup of open issues older than one year.

Decision: Closed: stale/non-critical; no evidence it is a critical current-version RNN issue.

If this still reproduces on the latest React Native Navigation with a minimal reproduction, please open a new issue with current RN/RNN versions and repro steps.
